resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/KubernetesClientBuilder.scala

```diff
@@ -17,11 +17,13 @@
 package org.apache.spark.deploy.kubernetes
 
 import java.io.File
+import java.util.concurrent.{ThreadFactory, ThreadPoolExecutor}
 
 import com.google.common.base.Charsets
 import com.google.common.io.Files
+import io.fabric8.kubernetes.client.utils.HttpClientUtils
 import io.fabric8.kubernetes.client.{Config, ConfigBuilder, DefaultKubernetesClient}
-
+import okhttp3.Dispatcher
```

Comment:
[Java kubernetes client API ](https://github.com/fabric8io/fabric8/tree/master/components/kubernetes-api) has a dependency on okhttp. okhttp is exposed  as the default Dispatcher constructor allows only non-daemon threads in their Executors, and we wanted to able to specify a custom dispatcher.